src/pages/githubAssessments/GitHubAssessmentWorkspace.tsx

```diff
@@ -0,0 +1,601 @@
+    const onChangeTabs = (
+      newTabId: SideContentType,
+      prevTabId: SideContentType,
+      event: React.MouseEvent<HTMLElement>
+    ) => {
+      if (newTabId === prevTabId) {
+        return;
+      }
+
+      // Do nothing when clicking the mobile 'Run' tab while on the autograder tab.
+      if (
+        !(prevTabId === SideContentType.autograder && newTabId === SideContentType.mobileEditorRun)
+      ) {
+        setSelectedTab(newTabId);
+      }
+    };
+    return {
+      mobileControlBarProps: {
+        ...controlBarProps()
+      },
+      ...sideContentProps(props),
+      onChange: onChangeTabs,
+      selectedTabId: selectedTab,
+      handleEditorEval: handleEval
+    };
+  };
+
+  const replButtons = () => {
+    const clearButton = (
+      <ControlBarClearButton handleReplOutputClear={props.handleReplOutputClear} key="clear_repl" />
+    );
+
+    const evalButton = (
+      <ControlBarEvalButton
+        handleReplEval={props.handleReplEval}
+        isRunning={props.isRunning}
+        key="eval_repl"
+      />
+    );
+
+    return [evalButton, clearButton];
+  };
+
+  const editorProps = {
+    editorSessionId: '',
+    editorValue: props.editorValue!,
+    handleDeclarationNavigate: props.handleDeclarationNavigate,
+    handleEditorEval: props.handleEditorEval,
+    handleEditorValueChange: onEditorValueChange,
+    breakpoints: props.breakpoints,
+    highlightedLines: props.highlightedLines,
+    newCursorPosition: props.newCursorPosition,
+    handleEditorUpdateBreakpoints: props.handleEditorUpdateBreakpoints,
+    handlePromptAutocomplete: props.handlePromptAutocomplete,
+    isEditorAutorun: false
+  };
+  const replProps = {
+    handleBrowseHistoryDown: props.handleBrowseHistoryDown,
+    handleBrowseHistoryUp: props.handleBrowseHistoryUp,
+    handleReplEval: props.handleReplEval,
+    handleReplValueChange: props.handleReplValueChange,
+    output: props.output,
+    replValue: props.replValue,
+    sourceChapter: selectedSourceChapter || 4,
+    sourceVariant: 'default' as Variant,
+    externalLibrary: ExternalLibraryName.NONE,
+    replButtons: replButtons()
+  };
+  const workspaceProps: WorkspaceProps = {
+    controlBarProps: controlBarProps(),
+    editorProps: editorProps,
+    editorHeight: props.editorHeight,
+    editorWidth: props.editorWidth,
+    handleEditorHeightChange: props.handleEditorHeightChange,
+    handleEditorWidthChange: props.handleEditorWidthChange,
+    handleSideContentHeightChange: props.handleSideContentHeightChange,
+    sideContentHeight: props.sideContentHeight,
+    sideContentProps: sideContentProps(props),
+    replProps: replProps
+  };
+  const mobileWorkspaceProps: MobileWorkspaceProps = {
+    editorProps: editorProps,
+    replProps: replProps,
+    mobileSideContentProps: mobileSideContentProps()
+  };
+
+  if (isLoading) {
+    return (
+      <div className={classNames('SideContentMissionEditor', Classes.DARK)}>
```

Comment:
i think this classname is misleading as this is not a sidecontent tab

Reply:
Renamed style:
https://github.com/chekjun/cadet-frontend/blob/8c389af6cb89646e2feb3eed084fbe405c06468f/src/pages/githubAssessments/GitHubAssessmentWorkspace.tsx#L545-L549